### DISCUSSION TITLE
Executable for linux_armv7 version does not work properly

### DISCUSSION THREAD

**Discussion by zuizui-kuroiko** [pocketbase/pocketbase#6132] [Q&A]
It appears that Pocketbase is not functioning correctly on Linux ARMv7. Versions up to v0.22.27 work without issues, but versions from v0.23.0 to v0.23.8 produce the following error.
```
$ ./pocketbase superuser create foo@bar.com password
2024/12/18 02:22:18 Failed to apply migration 1640988000_init.go: _mfas error: id: must be in a valid format.
```
Also, when trying to run pb_data that was used with Pocketbase v0.22.27 in Pocketbase v0.23.x, the following error occurs.
```
./pocketbase serve
2024/12/18 02:40:54 Failed to apply migration 1717233556_v0.23_migrate.go: failed to migrate admins->superusers: id: must be in a valid format.
```
I have confirmed this on a Raspberry Pi 3 running Raspbian (bullseye) 32-bit.

**Comment by ganigeorgiev** [comment]
I don't have access to armv7 device to test it.

Note that v0.23.0 comes with many breaking changes and requires manual migration steps. Please check the related changelog - https://github.com/pocketbase/pocketbase/releases/tag/v0.23.0 and more specifically follow the linked Go or JSVM upgrade guides if you are upgrading an older project.

Based on the error I suspect that it is possible to be something related to the crc32 checksum generation that we use for the collections id but I'm not able to debug it at the moment.


I'll try sometime later this week to setup a VM and will attempt to reproduce it but If you want to check it on your own you can try to add for example `println(col.Id)` in https://github.com/pocketbase/pocketbase/blob/master/migrations/1640988000_init.go#L155 to see what value it has generated and why it is incorrect (_to build PocketBase on your own see https://github.com/pocketbase/pocketbase#building-and-running-the-repo-maingo-example_).

**Reply by ganigeorgiev** [reply]
I was able to reproduce it by running the binary with QUEMU. It is caused by the 64->32bit int cast when calling `strconv.Itoa` which results to negative integer and the `-` character which is invalid for collection id. I'll try to push a fix later today or tomorrow.

**Reply by zuizui-kuroiko** [reply]
Thank you very much. I appreciate your prompt response.

**Reply by ganigeorgiev** [reply]
I've pushed a small update in [v0.23.9](https://github.com/pocketbase/pocketbase/releases/tag/v0.23.9) that should resolve the above error.

Ideally we should configure the GitHub action to run the automated tests against all platforms to prevent this from happening again but I'm not sure how exactly to do it (maybe custom VM inside the action's one?) and I'll have to research how other repos are doing it, but for now this remains a low priority (suggestions for the action config are welcomed).

**Reply by mroxso** [reply]
> I've pushed a small update in [v0.23.9](https://github.com/pocketbase/pocketbase/releases/tag/v0.23.9) that should resolve the above error.
> 
> Ideally we should configure the GitHub action to run the automated tests against all platforms to prevent this from happening again but I'm not sure how exactly to do it (maybe custom VM inside the action's one?) and I'll have to research how other repos are doing it, but for now this remains a low priority (suggestions for the action config are welcomed).
> 

Thank you for the fast fix! I was not affected by this issue but I stumbled across this and I think I got an idea on how to test this in the GitHub Action.
Maybe I will create a PR for it if I find some free time for it.
